crowbar_framework/app/models/service_object.rb

```diff
@@ -1576,4 +1616,12 @@ def profile(name, &block)
       block.call
     end
   end
+
+  def equal_or_match(value, maybe_regexp)
```

Comment:
Make the constraints more flexible without regular expressions.

* /regexp/
* >=
* >
* <=
* <
* ==

```ruby
  def equal_or_match(value, maybe_regexp)
    case
    when maybe_regexp.start_with?('/') && maybe_regexp.end_with?('/')
      Regexp.new(maybe_regexp[1..-2]).match(value)
    when maybe_regexp.start_with?('>')
      op, check = maybe_regexp.split(" ", 2)
      value.to_f.send(op.to_sym, check.to_f)
    when maybe_regexp.start_with?('>=')
      op, check = maybe_regexp.split(" ", 2)
      value.to_f.send(op.to_sym, check.to_f)
    when maybe_regexp.start_with?('<')
      op, check = maybe_regexp.split(" ", 2)
      value.to_f.send(op.to_sym, check.to_f)
    when maybe_regexp.start_with?('<=')
      op, check = maybe_regexp.split(" ", 2)
      value.to_f.send(op.to_sym, check.to_f)
    else
      value == maybe_regexp
    end
  end
```